### ISSUE TITLE
Log failed URL

### ISSUE THREAD

**Issue by jonnyzzz** [michel-kraemer/gradle-download-task#127]
The task failed the following way for me. It is not clear what exact URL it tried to process. Could you please consider wrapping all exceptions and adding or logging more detailed information on the failure. 

```
       Caused by: org.gradle.api.UncheckedIOException: org.apache.http.client.ClientProtocolException: Service Temporarily Unavailable
       	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:57)
       	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:40)
       	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:76)
       	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
       	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
       	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
       	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:801)
       	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:768)
       	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
       	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
      
       	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
       	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
       	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
       	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
       	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
       	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
       	... 100 more
       Caused by: org.apache.http.client.ClientProtocolException: Service Temporarily Unavailable
       	at de.undercouch.gradle.tasks.download.DownloadAction.openConnection(DownloadAction.java:400)
       	at de.undercouch.gradle.tasks.download.DownloadAction.executeHttpProtocol(DownloadAction.java:202)
       	at de.undercouch.gradle.tasks.download.DownloadAction.execute(DownloadAction.java:176)
       	at de.undercouch.gradle.tasks.download.DownloadAction.execute(DownloadAction.java:127)
       	at de.undercouch.gradle.tasks.download.Download.download(Download.java:64)
       	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
       	... 113 more
```

**Comment by michel-kraemer** [comment]
This is very similar to #115. I think I can implement this. I will check that and get back to you.

**Comment by btimofeev** [comment]
I have a similar problem. The plugin downloads files perfectly on my computer, but on the F-Droid build server the task ends with an error without specifying the address of the file being downloaded.

```
> Task :app:downloadDependencies FAILED

FAILURE: Build failed with an exception.

* Where:
Build file '/home/vagrant/build/org.emunix.insteadlauncher/app/build.gradle' line: 137

* What went wrong:
Execution failed for task ':app:downloadDependencies'.
> Could not download file
```

My issue in f-droid repo https://gitlab.com/fdroid/fdroiddata/issues/1624

**Comment by michel-kraemer** [comment]
This issue will be fixed in the upcoming version 4.0.0 of gradle-download-task
